crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1221,6 +1241,83 @@ impl World {
         result
     }
 
+    /// Temporarily removes the requested `!Send` resource from this [`World`], then re-adds it before
+    /// returning.
+    ///
+    /// This enables safe simultaneous mutable access to both a resource and the rest of the [`World`].
+    /// For more complex access patterns, consider using [`SystemState`](crate::system::SystemState).
+    ///
+    /// # Example
+    /// ```
+    /// use bevy_ecs::prelude::*;
+    /// #[derive(Resource)]
+    /// struct A(u32);
+    /// #[derive(Component)]
+    /// struct B(u32);
+    /// let mut world = World::new();
+    /// world.insert_resource(A(1));
+    /// let entity = world.spawn(B(1)).id();
+    ///
+    /// world.resource_scope(|world, mut a: Mut<A>| {
+    ///     let b = world.get_mut::<B>(entity).unwrap();
+    ///     a.0 += b.0;
+    /// });
+    /// assert_eq!(world.get_resource::<A>().unwrap().0, 2);
+    /// ```
+    pub fn non_send_scope<R: 'static, U>(&mut self, f: impl FnOnce(&mut World, Mut<R>) -> U) -> U {
```

Comment:
Rather than incur the complexity and safety risk of duplicating this impl, maybe we just remove it and see if anyone actually wants it? If we follow Joy's plan, it will probably be removed anyway as non send resources will be separate from World. And remove_non_send can be used in place of it if someone _really_ needs it (provided they are ok with incurring change detection events).

Reply:
I'm not sure if ripping the bandaid off now vs later is more preferable. I'd rather users make a small change here, and pay the cost of migrating away from this only when the storage for them changes.

Reply:
I've left the `resource_scope` code as is. That can be handled in a separate PR IMO. However, I've removed `non_send_scope` and it's tests.